File: src/main/java/no/nav/aap/api/søknad/domain/Fødselsnummer.java
package no.nav.aap.api.søknad.domain;

import com.fasterxml.jackson.annotation.JsonValue;

import javax.validation.constraints.NotNull;

import static no.nav.aap.api.søknad.util.StringUtil.partialMask;

public record Fødselsnummer(@NotNull @JsonValue String fnr) {

    @JsonValue(false)
    @Override
    public String fnr() {
        return fnr;
    }
    @Override
    public String toString() {
        return getClass().getSimpleName() + " [fnr=" + partialMask(fnr) + "]";
    }
}

File: src/main/kotlin/no/nav/aap/api/felles/Fødselsnummer.kt
package no.nav.aap.api.felles

import com.fasterxml.jackson.annotation.JsonValue
import no.nav.aap.api.util.StringUtil.partialMask

data class Fødselsnummer(@JsonValue val fnr: String) {
    override fun toString() = partialMask(fnr)
}